### DISCUSSION TITLE
A kanban view ?

### DISCUSSION THREAD

**Discussion by neo7-studio-web** [directus/directus#10380] [Archive]
Is it an upcoming feature ?

![tasks](https://user-images.githubusercontent.com/18073828/145106937-c96c749d-a40e-4553-90c3-d5bf1aaf1b85.jpg)

**Comment by benhaynes** [comment]
This actually exists!! This is our new Kanban Layout... but for now it's our first/only Cloud Exclusive extension! In time we might transition these exclusives back into the OSS community though!

**Reply by neo7-studio-web** [reply]
OK that's fair to have exclusive extensions for Cloud users. Maybe in a future marketplace there could be an option to buy some valuable extensions like this ? Just a suggestion. Thanks for your answer.

**Reply by rijkvanzanten** [reply]
That's the plan! 😄 🚀

**Reply by jdamboeck** [reply]
Oh dear. Sad direction directus is turning.

**Reply by benhaynes** [reply]
> Oh dear. Sad direction directus is turning.

How so? Our entire core platform is completely free, with no paywalls or limitations. No one pays us money for that, so there's no sustainable business there to keep the company running. 

We're building additional _extensions_ (in addition to the core capabilities) that we're planning to charge for as a business model. How else do we pay our staff so we can keep building/maintaining this open source software? Honest question... I'd love ideas.

**Comment by 3615alexis** [comment]
Hello,

Sorry to extract this subject 3 months after. But do you have any news regarding the release of Kanban view in Directus open source ?

This feature is quite standard in other open source solutions similar to Directus. For instance, NocoDB provides Kanban natively : https://www.nocodb.com/ (https://www.youtube.com/watch?v=GSqltFnj8So)

**Reply by benhaynes** [reply]
The Kanban Layout will be a Cloud Exclusive for the time being.

**Reply by 3615alexis** [reply]
That's clearer now. thanks ;)

**Reply by sxsws** [reply]
What other features are planned to be "cloud exclusive"? Do you plan to strip away any features to make them cloud exclusive? What extensions are planned / expected that will be commercial-only?

At what point will the base development of directus stop and features will be added by commercial extensions?

Is there a road map?

**Reply by benhaynes** [reply]
Nothing that is in Core will suddenly "switch" to be paid/exclusive, and no **key features** (this is subjective, but you'll just have to trust us) will be limited or paywalled. The best way that I can describe it is that we'll continue adding "icing on the cake" features to Cloud as exclusives. The reason for this is to...

1. Further differentiate our commercial offering
2. Which then drives more revenue to our company
3. Which then allows us to grow our Core team and resources
4. Which ultimately allows us to build a better open-source core product for you

_Lather, rinse, repeat..._

Base development will not stop, as a platform like Directus is never "complete". However, we are switching our focus to stability, docs, guides, and tests for the short-term (I say despite having Data Flows and other huge feature releases planned for the next month or so).

Also, we do keep a [high-level roadmap](https://directus.io/road-map/).

**Reply by sxsws** [reply]
Can you give a list of what are currently 'key features' and what are currently 'icing on the cake' so users can develop in such a way so they don't rely on features that will be removed. Odoo too removed the icing on the cake and stripped it. What remains is now a useless, but pretty-looking CMS that needs modules to be purchased to actually use it. The open source project now is just, if you will, a (literally) useless demo for the actual product/s - the modules/extensions. Their sales team tell you how much you will have to pay to use their "free" open source project right from the start.

If it's icing on the cake, how have you determined that they will actually drive more people to your commercial offering who are interested in your core offering? As I still see it from a user's pov, the insights module looks half done with basic options and can be replaced by more mature projects like Metabase, Redash or Superset (pulling direct from the database and with query builders for the sql challenged) and what with the API nor ui supporting things like views, Hasura and NoCoDB are close competition. Likewise, the more mature node-red, n8n or any other similar commercial IFTTT projects could almost be a drop-in replacement for data flows which seems like a severe time-and-code-hog at this point in development but is, along with insights, I'm sure great for demos to potential investors. Triggers, stored procedures and user-defined functions really add power to databases.

The no-code sector is a fiercely competitive area with new projects popping up everywhere. I just randomly came across motor admin yesterday that is way ahead of some things that directus isn't.

Not being able to create indexes mean that direct db interaction is necessary, so one may as well issue `CREATE TABLE` statements, or if the person doesn't know how to do this (because they're basic users who don't understand about table indexes), they will have a mass of slow tables which are actually doing them an injustice rather than a help. e.g. some old yahoo research ascertained that slower e-commerce sites turn customers away than faster sites. No tabs for forms mean putting say, 60 inputs on one long form. Even with an accordian click to open, scrolling down is still not fun.

What I'm trying to say is to concentrate on the basic, core product that makes directus the best project for whatevr the core purpose of directus is. Insights and dataflows muddies the water somewhat. Don't get lost by attempting to be everything for every person in a bid to increase audience size and put the core use of directus to one side in favour of half-finished time-hog features. Instead make it the best possible project so someone who wants to create 'collections' / 'tables' from a ui to suit their needs has no where else to go. 

If they then want to 'understand' or enhance their data using insights or dataflows for more turnover/profit, then that's where the people who will pay, will do so.

**Reply by benhaynes** [reply]
As I said, **nothing will be removed.** If it's built into Core now or in the future, it will stay there. Cloud exclusives will only come from new extensions being added in the future. And unfortunately, I can't predict which way new features will go beyond what I described above.

Regarding the rest of your message, we do not aim to be OK at _everything_, we aim to be a premium solution for a smaller set of use-cases. That said, no matter how you look at it, we're a small team and only have so many resources at the moment. The goal is to build out this project's skeleton first, which includes things like the data studio, settings config, insights, flows, asset management, etc. Once we have the bones in place (which we pretty much do at this point) we'll be focusing on developer experience for core + extensions.

I guess my point is that the priority of all features is subjective. You feel that Views, Query Builders, and Stored Procedures are crucial for Directus... others say that Flows/Insights are. We base our decisions on the votes within GitHub Discussions, as that is the best way we have to identify what the community wants next. In the end, things will all move faster once we have more resources available. Until then, we're doing what we can with what we have. :)

**Reply by sxsws** [reply]
Thanks for the clarification. However, there are some 18,000 stars on directus. Upvotes on discussions Don't even come close to 0.01% of its possible user base. You don't need a github account to be able to clone it so there will be users 'unable' to vote. I used to run a website with a healthy visitor count who would ask for bizarre features for the site! Such messages and requests were temporal and dependent on the feeling de jour rather than seeing the bigger picture.

I feel that database-level features are more important as a base, not particular features.

I think Directus is awesome and am trying to help out where I can.

**Reply by benhaynes** [reply]
> Thanks for the clarification. However, there are some 18,000 stars on directus. Upvotes on discussions Don't even come close to 0.01% of its possible user base. You don't need a github account to be able to clone it so there will be users 'unable' to vote.

Our ideal/primary customer profile is a developer... and I would argue that most devs have a GitHub account (even if they prefer BitBucket, GitLab, etc). GitHub is where we host our codebase, and is our authoritative place for organizing feature requests... that's why I mention it as the voting platform for us.

> I used to run a website with a healthy visitor count who would ask for bizarre features for the site! Such messages and requests were temporal and dependent on the feeling de jour rather than seeing the bigger picture.

Exactly! Everyone (rightfully) has their own opinions on this... so it's very subjective. While I agree with you that database-level features are most important, we have to also listen to the community and our corporate sponsors in what they request and vote on.

In the end, we do intend to focus on a deeper SQL/DB connection and the raw data tools... but all this takes time.

> I think Directus is awesome and am trying to help out where I can.

❤️

**Reply by vitalijalbu** [reply]
i really want **kanabn** view into directus for free...

**Comment by awesome-developer-20** [comment]
Can you please move this to open source?

It's really helpful for lot's of projects and I also want to look at it's architecture (the way it's implemented in directus).

It's a humble request and it's okay if it's not possible.

Really appreciate your work 🥰

Thanks 🙏

**Reply by Nitwel** [reply]
We will definitely make that open source in the future but probably not before we have something else that puts the icing on the cake in cloud. 👍

**Reply by awesome-developer-20** [reply]
Any eta?

Because it's been almost a year since this discussion

**Reply by benhaynes** [reply]
No ETA or plans to make this open source as of now.

**Reply by dangerstudios** [reply]
plus 100 for the opensource kanban! or... a tut to make a kanban extension :)

**Reply by chouchua** [reply]
upvote

**Comment by liuxiaodong008008** [comment]
Also really need the kanban layout in oss.   Looking forward to the release. 🥇

**Comment by agavronek** [comment]
Any way to purchase/license the exclusive kanban layout to be included on self-hosted environments?

Working on building a backoffice tool based on directus. Based on our requirements we really need to self-host, as we would like to roll a bunch of custom extensions (integrations/sync mechanisms to external private services, custom UIs in some cases). However some sort of project management is part of the backoffice (as often the case), so I am not looking forward to reimplement a Kanban layout:) Any options by any chance?

**Reply by Ppizarrojr** [reply]
Kanban is available as part of our cloud exclusive extensions, as well as our "Self-Hosted Enterprise" license. Feel free to send an email to sales@directus.io and we can provide pricing and features if that's what you're looking for.

**Comment by jramiroz98** [comment]
Maybe you should add a sidenote when implementing it. I was a bit frustrated when I couldn't find the Kanban View as I was moving to my own deployment of directus. But, like you mentioned above it's just one feature, everything else is still free and it's one hell of a tool you are sharing with the community for free, so I understand.

**Comment by FredericLatour** [comment]
Maybe you could sell a package of custom extensions. 
The cloud offering does not work in many cases for small biz (as soon as you need a custom extension which is a very common need).
However, I'm sure many people would accept to pay a reasonable price for some additional features.

**Comment by 53RG1005** [comment]
If anyone is looking for an alternative, I found the perfect one

https://github.com/shocota/directus-extension-board-layout

**Reply by dangerstudios** [reply]
yes a kanban extension, awesome!

**Comment by almereyda** [comment]
Directus now has a native Kanban layout, which is very similar to the ideas presented in the previous extension.

* [Kanban Layout | Layouts | Directus Docs](https://docs.directus.io/user-guide/content-module/layouts.html#kanban-layout)

This was released with [`v10.1.0`](https://github.com/directus/directus/releases/tag/v10.1.0).
